File: support/src/main/java/com/wuguangxin/adapter/base/BaseViewAdapter.java
package com.wuguangxin.adapter.base;

import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * View的适配器
 * Created by wuguangxin on 2019/12/2.
 */
public class BaseViewAdapter extends PagerAdapter {
    private List<View> list;

    public BaseViewAdapter(List<View> list) {
        this.list = list;
    }

    @Override
    public int getCount() {
        return list == null ? 0 : list.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
        return o == view;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        if (list == null)
            return null;
        container.addView(list.get(position));
        return list.get(position);//View

    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(list.get(position));
    }

}

File: support/src/main/java/com/wuguangxin/adapter/BaseViewAdapter.kt
package com.wuguangxin.adapter

import android.view.View
import androidx.viewpager.widget.PagerAdapter
import android.view.ViewGroup

/**
 * View的适配器
 * Created by wuguangxin on 2019/12/2.
 */
class BaseViewAdapter(val list: MutableList<View>?) : PagerAdapter() {
    private val mData = mutableListOf<View>()

    init {
        this.mData.clear()
        list?.let {
            this.mData.addAll(list)
        }
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return o === view
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = mData[position]
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mData[position])
    }

    override fun getCount(): Int {
        return mData.size
    }
}